From: Ralf Baechle <ralf@linux-mips.org>
Date: Tue, 3 Oct 2006 13:42:02 +0000 (+0100)
Subject: [MIPS] BCM1480: Mask pending interrupts against c0_status.im.
X-Git-Tag: linux-1.1.68
X-Git-Url: http://www.linux-mips.org/git?p=linux.git;a=commitdiff;h=b1f4fee6a18b48da275920df71b312e01ba973b9

[MIPS] BCM1480: Mask pending interrupts against c0_status.im.

Signed-off-by: Ralf Baechle <ralf@linux-mips.org>
(cherry picked from f2268c74623b9d82e4fe9b1e0951706c2360109e commit)
---

--- a/arch/mips/sibyte/bcm1480/irq.c
+++ b/arch/mips/sibyte/bcm1480/irq.c
@@ -482,7 +482,7 @@ asmlinkage void plat_irq_dispatch(struct
 	write_c0_compare(read_c0_count());
 #endif
 
-	pending = read_c0_cause();
+	pending = read_c0_cause() & read_c0_status();
 
 #ifdef CONFIG_SIBYTE_BCM1480_PROF
 	if (pending & CAUSEF_IP7)	/* Cpu performance counter interrupt */
